PythonVisualizations/SimpleQueue.py

```diff
@@ -230,133 +264,127 @@ def display(self):
-
-def onClick(command, parameter = None):
-    cleanUp()
-    #disableButtons()
-    if parameter:
-        command(parameter)
-    else:
-        command()
-    #if command not in [pause, play]:
-     #   enableButtons()
-
-def cleanUp():
-    global cleanup
-    if len(cleanup) > 0:
-        for o in cleanup:
-            canvas.delete(o)
-    outputText.set('')
-    window.update()
+    def onClick(self,command, parameter = None):
+        self.cleanUp()
+        #disableButtons()
+        if parameter:
+            command(parameter)
+        else:
+            command()
+        #if command not in [pause, play]:
+         #   enableButtons()
 
 # Button functions
 
-def clickInsertRear():
-    entered_text = textBox.get()
-    if entered_text:
-        val = int(entered_text)
-        if val < 100:
-            queue.insertRear(int(entered_text))
-        else:
-            outputText.set("Input value must be an integer from 0 to 99.")
-        textBox.delete(0, END )
+    def clickInsertRear(self):
+        entered_text = self.textEntries[0].get()
+        if entered_text:
+            val = int(entered_text)
+            if val < 100:
+                self.insertRear(int(entered_text))
+            else:
+                self.outputText.set("Input value must be an integer from 0 to 99.")
+            self.textEntries[0].delete(0, END )
         
-def clickInsertFront():
-    entered_text = textBox.get()
-    if entered_text:
-        val = int(entered_text)
-        if val < 100:
-            queue.insertFront(int(entered_text))
-        else:
-            outputText.set("Input value must be an integer from 0 to 99.")
-        textBox.delete(0, END )
+    def clickInsertFront(self):
+        entered_text = self.textBox.get()
+        if entered_text:
+            val = int(entered_text)
+            if val < 100:
+                self.insertFront(int(entered_text))
+            else:
+                self.outputText.set("Input value must be an integer from 0 to 99.")
+            self.textBox.delete(0, END )
         
-def clickEnableQueue():
-    #THIS IS HARDWIRED (THE ORDER OF THE BUTTONS) - FIX
-    buttons[1].config(state = DISABLED) 
-    buttons[2].config(state = DISABLED)
-    
-    # Toggling between deque and queue
-    buttons[4].config(relief = SUNKEN)
-    buttons[5].config(relief = RAISED)
-    
-    queue.onOffButtons()
-    
-def clickEnableDeque():
-    for button in buttons:
-        button.config(state = NORMAL)
-    
-    # Toggling between deque and queue
-    buttons[4].config(relief = RAISED)
-    buttons[5].config(relief = SUNKEN)
+    def clickEnableQueue(self):
+        #THIS IS HARDWIRED (THE ORDER OF THE BUTTONS) - FIX
+        self.buttons[1].config(state = DISABLED)
+        self.buttons[2].config(state = DISABLED)
+
+        # Toggling between deque and queue
+        self.buttons[4].config(relief = SUNKEN)
+        self.buttons[5].config(relief = RAISED)
```

Comment:
The comment on Line 328 is correct.  Instead of referring to operation buttons by an index in a list of buttons, the code is much easier to read and maintain when named attributes are used.  So instead of `self.buttons[1]`, for example, use `self.insertFrontButton`